modules/nw-dns-forward.adoc

```diff
@@ -42,10 +43,24 @@ spec:
       upstreams:
         - 3.3.3.3
         - 4.4.4.4:5454
+  upstreamResolvers: <4>
+      policy: Random <5>
+      upstreams: <6>
+        - type: SystemResolvConf <7>
+        - type: Network
+          address: 1.2.3.4 <8>
+          port: 53 <9>
 ----
 <1> `name` must comply with the `rfc6335` service name syntax.
 <2> `zones` must conform to the definition of a `subdomain` in `rfc1123`. The cluster domain, `cluster.local`, is an invalid `subdomain` for `zones`.
 <3> A maximum of 15 `upstreams` is allowed per `forwardPlugin`.
+<4> `upstreamResolvers` is optional. You can use it to override the default policy and proxy DNS messages to the specified DNS resolvers (upstream resolvers) for the default domain. If you do not provide any upstream resolvers, the DNS name queries resolve to `/etc/resolv.conf`.
+<5> `policy` determines the order in which upstream servers are selected for querying. You can specify one of these values: `Random`, `RoundRobin`, and `Sequential`. The default value is `Sequential`.
+<6> `upstreams` is optional. You can use it to provide upstream resolvers.
+<7> You can specify two types of `upstreams` - `SystemResolvConf` and `Network`. `SystemResolvConf` is for configuring the upstream to use `/etc/resolv.conf` and `Network` is for defining a `Networkresolver`. You can specify one or both.
+<8> If the specified type is `Network`, you must provide an IP address. `address` must be a valid IPv4 or IPv6 address.
+<9> If the specified type is `Network`, you can optionally provide a port. `port` must be between 1 and 65535.
+
```

Comment:
Perfect @chinmayi-chandrasekar , all is accurate here. :+1: 
I'm a bit concerned that it seems rather complex to understand for our readers, having many fields optional at several levels. 
Aspects such as mandatory vs optional, allowed values, etc would be in https://docs.openshift.com/container-platform/4.8/rest_api/operator_apis/dns-operator-openshift-io-v1.html (or equivalent)
So maybe in this page, we could give more simple and practical examples, such as:
* I want to override the DNS servers for a specific zone
* I want to override the default DNS server (the upstream) and use a list of network addresses
* I want to keep /etc/resolv.conf but use a different policy
* maybe only as note, specify that it is possible to mix both
* I